core/modules/cart/helpers/productsEquals.ts

```diff
@@ -34,10 +57,19 @@ const productsEquals = (product1: CartItem, product2: CartItem): boolean => {
   const typeProduct1 = getProductType(product1)
   const typeProduct2 = getProductType(product2)
 
+  const hasProductCustomOption1 = hasCustomOptions(product1)
+  const hasProductCustomOption2 = hasCustomOptions(product2)
+
   if (typeProduct1 === 'bundle' || typeProduct2 === 'bundle') {
     return isServerIdsEquals(product1, product2) || isChecksumEquals(product1, product2)
   }
 
+  if (hasProductCustomOption1 && hasProductCustomOption2) {
```

Comment:
Shouldn’t it be done exactly the same way like bundle products comparison above? Product custom options and bundle options works almost the same way and I guess using checksums is a much cleaner approach. The check above was limited to just bundles and custom options can be assigned to any type of product so I guess that maybe just removing this product.type_id check and  leaving the checksum check for all type of products could do the work?

I might be wrong so please let me know if that’s a case :-)